thunder/executors/cudagraphex.py

```diff
@@ -138,6 +138,10 @@ def fuse(self, region: Region, fusion_counter: int, num_static_inputs: None | in
         inputs = [unvariableify(inp) for inp in sorted(region.inputs, key=lambda var: var.proxy.name)]
         outputs = [unvariableify(out) for out in sorted(region.outputs, key=lambda var: var.proxy.name)]
 
+        from thunder.executors.passes import _del_last_used
+
+        region.bound_symbols = _del_last_used(region.bound_symbols, outputs)
```

Comment:
Does Thunder need region transforms?

What's the benefit of having `del` calls at all inside the CUDA Graphed region? CUDA Graphs are used to escape Python and `del` is a Python-only concept.

Reply:
Would we not need them for memory reuse inside the cudagraph?

Reply:
OK, I'll remove these bits then. :)

Reply:
Thinking about it more, I am still confused here. For very large graphs we would be glad to re-use memory, I think, and I would hope to achieve this by `del`ing during capture...